ocs_ci/ocs/ui/page_objects/bucket_tab_permissions.py

```diff
@@ -0,0 +1,692 @@
+
+
+class BucketsTabPermissions(ObjectStorage, ConfirmDialog):
+    """
+    A class representation for abstraction of Buckets tab permissions related OpenShift UI actions
+    """
+
+    # Policy action button keys to try in order
+    _POLICY_ACTION_BUTTONS = ("apply_policy_button", "save_policy_generic_button")
+
+    def navigate_to_bucket_permissions(self, bucket_name: str = None) -> None:
+        """
+        Navigate to bucket permissions tab.
+
+        Args:
+            bucket_name (str, optional): Name of the bucket. If None, uses first bucket.
+
+        Raises:
+            NoSuchElementException: If UI elements are not found.
+        """
+        if not bucket_name:
+            self.do_click(self.bucket_tab["bucket_list_items"])
+            self.do_click(self.bucket_tab["permissions_tab"])
+            self.do_click(self.bucket_tab["bucket_policy_tab"])
+            return
+
+        bucket_elements = self.get_elements(self.bucket_tab["bucket_list_items"])
+
+        for bucket_element in bucket_elements:
+            if bucket_element.text != bucket_name:
+                continue
+
+            bucket_element.click()
+            break
+        else:
+            available_buckets = [elem.text for elem in bucket_elements]
+            raise NoSuchElementException(
+                f"Bucket '{bucket_name}' not found in bucket list. "
+                f"Available buckets: {available_buckets}. "
+                "Verify bucket name exists and is visible on current page."
+            )
+
+        self.do_click(self.bucket_tab["permissions_tab"])
+        self.do_click(self.bucket_tab["bucket_policy_tab"])
+
+    def activate_policy_editor(self) -> None:
+        """
+        Activate the policy editor by intelligently choosing between 'Edit policy' and 'Start from scratch'.
+
+        This method checks if a policy already exists on the bucket:
+        - If policy exists: clicks 'Edit policy' button
+        - If no policy exists: clicks 'Start from scratch' button
+
+        Raises:
+            NoSuchElementException: If UI elements are not found.
+        """
+        try:
+            self.wait_for_element_to_be_visible(
+                self.bucket_tab["edit_policy_button"], timeout=QUICK_WAIT
+            )
+            self.do_click(self.bucket_tab["edit_policy_button"])
+            return
+        except (NoSuchElementException, TimeoutException):
+            pass
+
+        try:
+            self.wait_for_element_to_be_visible(
+                self.bucket_tab["policy_editor_start_scratch"], timeout=QUICK_WAIT
+            )
+            self.do_click(self.bucket_tab["policy_editor_start_scratch"])
+            return
+        except (NoSuchElementException, TimeoutException):
+            pass
+
+        raise NoSuchElementException(
+            "Could not find either 'Edit policy' or 'Start from scratch' button. "
+            "Check if bucket permissions page is properly loaded."
+        )
```

Comment:
why this method allows `TimeoutException`? In this case there is no much sense to make "wait" for `edit_policy_button` or `policy_editor_start_scratch` as far as I understand.